http-server/src/main/java/com/facebook/airlift/http/server/HttpServer.java

```diff
@@ -427,9 +431,19 @@ private static ServletContextHandler createServletContext(
         context.addServlet(servletHolder, "/*");
 
         for (Map.Entry<String, Servlet> servlet : servlets.entrySet()) {
-            ServletHolder holder = new ServletHolder(servlet.getValue());
-            holder.setInitParameters(ImmutableMap.copyOf(parameters));
-            context.addServlet(holder, servlet.getKey());
+            if (config.isAuthEnabled()) {
```

Comment:
It feels like naturally, if `Authorizer` is present the authorization should be enabled. If `Authorizer` is missing then it should be disabled. Otherwise it's a little not intuitive when you pass an authorizer, but the authorization is not enabled.

Reply:
@arhimondr - one usecase I can think of is that the authorizer is present in code but we want to disable authorization through a config. That said, it can be done on the implementation side as well that if the config is disabled, then we don't bind the authorizer.